### ISSUE TITLE
Dashboard relative date filter widget doesn't work with SQL card field filter variables

### ISSUE THREAD

**Issue by hai-ld** [metabase/metabase#3538]
For a (simplified) example query as follows:

```sql
SELECT CAST("public"."user_account"."created_at" AS date) AS "created_at", count(*) AS "count"
FROM "public"."user_account"
GROUP BY CAST("public"."user_account"."created_at" AS date)
ORDER BY CAST("public"."user_account"."created_at" AS date) ASC
```

If it's constructed via query builder, when I add the question to dashboard and add a relative time filter, it would be able to recognize created_at as a compatible field. But if it's a custom SQL question, created_at would not be shown.

- Your browser and the version: Chrome Version 53.0.2785.143 (64-bit)
- Your operating system: Ubuntu 16.04
- Your databases: Postgres
- Metabase version: 0.20.0 and 0.19.3
- Metabase hosting environment: Docker
- Metabase internal database: H2

**Comment by tlrobinson** [comment]
It would be very difficult for us to do this automatically, but we do have a way to manually annotate your SQL with fields you'd like to be able to filter by: http://www.metabase.com/docs/latest/users-guide/12-sql-parameters In particular you'll want to use the "Field filter" variable type.

**Comment by hai-ld** [comment]
@tlrobinson I tried to modify the question as 
```sql
SELECT CAST("public"."user_account"."created_at" AS date) AS "created_at", count(*) AS "count"
FROM "public"."user_account"
WHERE {{created_date}}
GROUP BY CAST("public"."user_account"."created_at" AS date)
ORDER BY CAST("public"."user_account"."created_at" AS date) ASC
```

but some how I got this
```
 :query
 {:type "native",
  :native
  {:query
   "SELECT CAST(\"public\".\"user_account\".\"created_at\" AS date) AS \"created_at\", count(*) AS \"count\"\nFROM \"public\".\"user_account\"\nWHERE {{created_date}}\nGROUP BY CAST(\"public\".\"user_account\".\"created_at\" AS date)\nORDER BY CAST(\"public\".\"user_account\".\"created_at\" AS date) ASC\n",
   :template_tags {:created_date {:id "4bd6f5d0-2cdc-721c-54cf-daca536bf0f7", :name "created_date", :display_name "Created date", :type "dimension", :dimension ["field-id" 2113]}}},
  :parameters [{:type "date/relative", :target ["dimension" ["template-tag" "created_date"]], :value "past7days"}],
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :info {:executed-by 4, :card-id 36, :uuid "7490b5a6-721d-4881-8d4c-97d55866e2bc", :query-hash 315105816, :query-type "native"}},
 :expanded-query nil}
10-12 08:09:57 ERROR metabase.query-processor :: Query failure: ERROR: invalid input syntax for type date: "past7days"
```

The query seems to become
```sql
SELECT CAST("public"."user_account"."created_at" AS date) AS "created_at", count(*) AS "count"
FROM "public"."user_account"
WHERE CAST("public"."user_account"."created_at" AS date) = 'past7days'
GROUP BY CAST("public"."user_account"."created_at" AS date)
ORDER BY CAST("public"."user_account"."created_at" AS date) ASC
```

I'm using 0.20. Wondering whether this is a regression?

**Comment by tlrobinson** [comment]
Actually you're right, but it looks like the relative date widget is broken for SQL cards / field filter variables in 0.19 as well.

The relative date widget works fine for query builder cards, and single/range/quarter/month widgets all work for both query builder and SQL cards, so it looks like it's just SQL + relative that doesn't work.

![screenshot 2016-10-12 01 49 38](https://cloud.githubusercontent.com/assets/18193/19303642/5757dc60-901e-11e6-9c41-b2f8e30947b8.png)
![screenshot 2016-10-12 01 49 52](https://cloud.githubusercontent.com/assets/18193/19303641/57549e74-901e-11e6-8ae9-94841d056cc1.png)

I'll reopen this and rename it to reflect the relative date + SQL card issue.

**Comment by camsaul** [comment]
Fixed by #3589
